packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -585,52 +572,21 @@ export function attach(
     }
   }
 
-  function onErrorOrWarning(
-    fiber: Fiber,
-    type: 'error' | 'warn',
-    args: $ReadOnlyArray<any>,
+  function onErrorsAndWarningsUpdate(
+    errorsAndWarnings: Array<{|
+      fiber: Fiber,
+      type: 'error' | 'warn',
+      args: $ReadOnlyArray<any>,
+    |}>,
   ): void {
-    // There are a few places that errors might be logged:
-    // 1. During render (either for initial mount or an update)
-    // 2. During commit effects (both active and passive)
-    // 3. During unmounts (or commit effect cleanup functions)
-    //
-    // Initial render presents a special challenge-
-    // since neither backend nor frontend Store know about a Fiber until it has mounted (and committed).
-    // In this case, we need to hold onto errors until the subsequent commit hook is called.
-    //
-    // Passive effects also present a special challenge-
-    // since the commit hook is called before passive effects, are run,
-    // meaning that we might not want to wait until the subsequent commit hook to notify the frontend.
-    //
-    // For this reason, warnings/errors are sent to the frontend periodically (on a timer).
-    // When processing errors, any Fiber that is not still registered is assumed to be unmounted.
-    pendingErrorOrWarnings.push({
-      fiber,
-      type,
-      args,
-    });
-
-    if (flushErrorOrWarningUpdatesTimoutID === null) {
-      flushErrorOrWarningUpdatesTimoutID = setTimeout(
-        flushErrorOrWarningUpdates,
-        1000,
-      );
-    }
-  }
-
-  function flushErrorOrWarningUpdates() {
-    flushErrorOrWarningUpdatesTimoutID = null;
-
     const updatedFiberIDs: Set<number> = new Set();
-    pendingErrorOrWarnings
+    errorsAndWarnings
       .filter(({fiber}) => isFiberMounted(fiber))
       .forEach(({args, fiber, type}) => {
```

Comment:
nit: We could get rid of the extra `filter` traversal by just moving the `isFiberMounted` check into the `forEach`.

Reply:
But how are people supposed to know you know functional programming?

Reply:
I think our use of the `forEach` method already killed that impression 😆